src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -6801,6 +6801,12 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_CollectionLiteralNoTargetType" xml:space="preserve">
     <value>There is no target type for the collection literal.</value>
   </data>
+  <data name="ERR_CollectionBuilderAttributeMethodNotFound" xml:space="preserve">
+    <value>Could not find an accessible '{0}' method with the expected signature.</value>
```

Comment:
It feels reasonable for the diagnostic message to specify the expected signature and modifiers. e.g. `'{candidateSymbol}' must be a static method with a single parameter of type ReadOnlySpan<T> to be used as a CollectionBuilder method`

Reply:
Updated the message to include the parameter and return types.

As we discussed, an alternative is to generate a `SignatureOnlyMethodSymbol` with the expected signature and rely on `SymbolDisplay`. I held off using that approach for now since that requires additional work to construct the method symbol and requires implementing some `SignatureOnlyMethodSymbol` properties that currently `throw ExceptionUtilities.Unreachable`.